theano/tensor/basic.py

```diff
@@ -3372,52 +3372,483 @@ def transpose(x, axes=None):
     return ret
 
 
-def batched_dot(x, y):
+class BatchedDot(Op):
```

Comment:
I would move that op to tensor/blas.py.
I would then use theano.tensor.blas.BatchedDot inside the batched_dot method in this file.